webapi/src/main/scala/org/knora/webapi/responders/v1/OntologyResponderV1.scala

```diff
@@ -462,4 +498,240 @@ class OntologyResponderV1 extends ResponderV1 {
+
+            case None => // map over all named graphs and collect the resource types
+                val resourceTypesFuture: Vector[Future[Vector[ResourceTypeV1]]] = settings.namedGraphs.filter(_.visibleInGUI).map {
+                    namedGraphs: ProjectNamedGraphs => getResourceTypes(namedGraphs.ontology)
+                }
+
+                val sequencedFuture = Future.sequence(resourceTypesFuture)
+
+                for {
+                    resourceTypes <- sequencedFuture
+                } yield ResourceTypesForNamedGraphResponseV1(resourcetypes = resourceTypes.flatten, userdata = userProfile.userData)
+        }
+
+    }
+
+    /**
+      * Gets the property types defined in the given named graph. If there is no named graph defined, get property types for all existing named graphs.
+      *
+      * @param namedGraphIriOption the Iri of the named graph or None if all the named graphs should be queried.
+      * @param userProfile         the profile of the user making the request.
+      * @return a [[PropertyTypesForNamedGraphResponseV1]].
+      */
+    private def getPropertyTypesForNamedGraph(namedGraphIriOption: Option[IRI], userProfile: UserProfileV1): Future[PropertyTypesForNamedGraphResponseV1] = {
+
+        def getPropertiesForNamedGraph(namedGraphIri: IRI, userProfile: UserProfileV1): Future[Vector[PropertyDefinitionV1]] = {
+            for {
+                namedGraphEntityInfo <- getNamedGraphEntityInfoV1ForNamedGraph(namedGraphIri, userProfile)
+                propertyIris: Vector[IRI] = namedGraphEntityInfo.propertyIris
+                entities: EntityInfoGetResponseV1 <- getEntityInfoResponseV1(propertyIris = propertyIris.toSet, userProfile = userProfile)
+                propertyEntityInfoMap: Map[IRI, PropertyEntityInfoV1] = entities.propertyEntityInfoMap
+
+                propertyDefinitions: Vector[PropertyDefinitionV1] = propertyEntityInfoMap.map {
+                    case (propertyIri: IRI, entityInfo: PropertyEntityInfoV1) =>
+
+                        if (entityInfo.isLinkProp) {
+                            // It is a linking prop: its valuetype_id is knora-base:LinkValue.
+                            // It is restricted to the resource class that is given for knora-base:objectClassConstraint
+                            // for the given property which goes in the attributes that will be read by the GUI.
+
+                            PropertyDefinitionV1(
+                                id = propertyIri,
+                                name = propertyIri,
+                                label = entityInfo.getPredicateObject(OntologyConstants.Rdfs.Label),
+                                description = entityInfo.getPredicateObject(OntologyConstants.Rdfs.Comment),
+                                vocabulary = entityInfo.predicates.values.head.ontologyIri,
+                                occurrence = "",
```

Comment:
A property's cardinality cannot be set in the context of a named graph because the cardinality is defined for the resource class, and a property might be used in more than one resource class in one named graph.